### ISSUE TITLE
What will be the proper hardware to run train.py

### ISSUE THREAD

**Issue by edisondeng** [karpathy/llama2.c#292]
hi all:

I tried to train the tinystories on my PC. but it seems the train.py is running very slow. It actually doesn't consume too much VRAM which is only 136MB from the output of nvidia-smi, but it did consume 100% cpu from the top command. My hardware is i7-12700H + RTX3060.  Any suggestions?

tks

**Comment by espio999** [comment]
Hello, this is my case.
Intel Core i5 (7th generation), 8GB RAM + 8GB swap, no GPU.
Process was killed in 40min as attached.

I changed 2 parameters below for this run as
device = "cpu"
compile = False

parameters mentioned above are default
max_seq_len = 256
batch_size = 128
n_layers = 6
n_heads = 6
dropout = 0.0
learning_rate = 5e-4

any suggestion except hardware enhancement/replacement?

![train py](https://github.com/karpathy/llama2.c/assets/8131672/d1a9e2a8-cb64-42ab-96e0-64f751af992b)
